Sources/Datadog/Tracing/DDSpanContext.swift

```diff
@@ -49,7 +49,7 @@ internal class BaggageItems {
     }
 
     func get(key: String) -> String? {
-        queue.sync { self.unsafeItems[key] }
+        queue.sync { self.unsafeItems[key] } ?? parent?.get(key: key)
```

Comment:
This read isn't atomic and blocks the thread several times:
```swift
queue.sync { ... } ?? queue.sync { ... } ?? queue.sync { ... } ?? ... N
```
depending on how many `N` parent contexts there are in the hierarchy.

---
We can make it atomic and ensure there's only one thread sync (`queue.sync { ... }`) executed if we just do:
```swift
queue.sync { self.unsafeAll[key] }
```

As `unsafeAll` doc comment says, it:
> Returns all baggage items for the span, including its parent items.